58BalanceBinaryTree.py

```diff
@@ -0,0 +1,72 @@
+"""
+// Time Complexity :O(n2)
+// Space Complexity :O(1)
```

Comment:
Space complexity isn't constant because we are using recursive stack space. It is O(H)

Reply:
I got it.